File: python/greeting_client.py
import logging

import grpc

from protos import greeting_pb2
from protos import greeting_pb2_grpc


def run():
    with grpc.insecure_channel("localhost:50051") as channel:
        stub = greeting_pb2_grpc.GreeterStub(channel)
        response = stub.SayHello(greeting_pb2.HelloReq(name="world"))
    print("Received: " + response.message)


if __name__ == "__main__":
    logging.basicConfig()
    run()

File: php/greeting_client.php
<?php

require dirname(__FILE__).'/vendor/autoload.php';

function greet($hostname, $name)
{
    $client = new Greeting\GreeterClient($hostname, [
        'credentials' => Grpc\ChannelCredentials::createInsecure(),
    ]);
    $request = new Greeting\HelloReq();
    $request->setName($name);
    list($response, $status) = $client->SayHello($request)->wait();
    if ($status->code !== Grpc\STATUS_OK) {
        echo "ERROR: " . $status->code . ", " . $status->details . PHP_EOL;
        exit(1);
    }
    echo $response->getMessage() . PHP_EOL;
}

$name = empty($argv[1]) ? 'world': $argv[1];
$hostname = 'localhost:5566';
greet($hostname, $name);
